client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+import os
+import sys
+import time
+import stat
+import shutil
+
+import king_phisher.client.plugins as plugins
+
+from gi.repository import Gtk
+from gi.repository.GdkPixbuf import Pixbuf
+
+import boltons.strutils
+import boltons.timeutils
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Josh Jacob']
+	title = 'SFTP Client GUI'
+	description = """
+	How many foos could fu foo is fu fooed fu foos on foo.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher'
+
+	def initialize(self):
+		module = sys.modules['king_phisher.client.application']
+		self.temp = module.KingPhisherClientApplication.do_sftp_client_start
+		module.KingPhisherClientApplication.do_sftp_client_start = self.new_SFTP_signal
+		sys.modules['king_phisher.client.application'] = module
+		return True
+
+	def _cleanup(self):
+		module = sys.modules['king_phisher.client.application']
+		module.KingPhisherClientApplication.do_sftp_client_start = self.temp
+		sys.modules['king_phisher.client.application'] = module
+
+	def new_SFTP_signal(self):
+		if not self.application._ssh_forwarder:
+			config = self.application.config
+			server = config['server']
+			server = server.split(':')
+			username = config['server_username']
+			password = config['server_password']
+			self.application._create_ssh_forwarder(server, username, password)
+		connection = self.application._ssh_forwarder
+		ssh = connection.client
+		ftp = ssh.open_sftp()
+
+		target_file = os.path.splitext(__file__)[0]
+		builder = Gtk.Builder()
+		builder.add_from_file(target_file + '.ui')
+		self.window = builder.get_object('SFTPClientGUI.window')
+		manager = FileManager(builder, self.application, ftp)  #pylint: disable=unused-variable
+		self.window.show_all()
+
+class Logger(object):
+	def __init__(self, builder, *args, **kwargs):
+		self.builder = builder
+		self.scroll = self.builder.get_object('scrolledwindow2')
+		self.treeview_transfer = self.builder.get_object('SFTPClientGUI.treeview_transfer')
+		self.progress_bar = self.builder.get_object('SFTPClientGUI.progressbar')
+		self.label_file = self.builder.get_object('SFTPClientGUI.label')
+		col = Gtk.TreeViewColumn('Transfer')
+		colText = Gtk.CellRendererText()
+		col.pack_start(colText, True)
+		col.add_attribute(colText, "text", 0)
+		self.treeview_transfer.append_column(col)
+		self._tvmodel = Gtk.ListStore(str)
+		self.treeview_transfer.connect('size-allocate', self._treeview_changed)
+		self.treeview_transfer.set_model(self._tvmodel)
+		self.treeview_transfer.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.MenuItem.new_with_label('Clear')
+		menu_item.connect('activate', self.clear_log)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == 3:
```

Comment:
Please include a comment as to what 3 is, or if there is a constant available in `Gdk` like `Gdk.BUTTON_SECONDARY`.